packages/flutter/lib/src/rendering/debug.dart

```diff
@@ -96,9 +96,18 @@ bool debugCheckIntrinsicSizes = false;
 
 /// Adds [dart:developer.Timeline] events for every [RenderObject] layout.
 ///
-/// For details on how to use [dart:developer.Timeline] events in the Dart
-/// Observatory to optimize your app, see:
-/// <https://fuchsia.googlesource.com/topaz/+/master/shell/docs/performance.md>
+/// The timing information this flag exposes is not representative of the actual
+/// cost of layout, because the overhead of adding timeline events is
+/// significant relative to the time each object takes to lay out. However, it
+/// can expose unexpected layout behavior in the timeline.
+///
+/// In debug builds, additional information is included in the trace (such as
+/// the properties of render objects being laid out). Collecting this data is
+/// expensive and further makes these traces non-representative of actual
+/// performance. This data is omitted in profile builds.
```

Comment:
Since the first version of this PR didn't do the "This data is omitted in profile builds.": Can we add a test to ensure that we don't regress this and accidentally include this in profile builds again?

Reply:
I ended up writing such a test by having test.dart compile an app in profile mode and then examining the build output for strings that would indicate that the method was incorrectly compiled in. I'm not 100% proud of the result, but it does seem to do what you want. :-)